torchvision/prototype/transforms/functional/_geometry.py

```diff
@@ -404,9 +404,12 @@ def _compute_affine_output_size(matrix: List[float], w: int, h: int) -> Tuple[in
 
 
 def _apply_grid_transform(
-    float_img: torch.Tensor, grid: torch.Tensor, mode: str, fill: datapoints.FillTypeJIT
+    img: torch.Tensor, grid: torch.Tensor, mode: str, fill: datapoints.FillTypeJIT
 ) -> torch.Tensor:
 
+    fp = img.dtype == grid.dtype
```

Comment:
Could you please explain why we're sure that img.dtype is float iff it's the same as the grid dtype? Can't we just use `is_floating_dtype()`?

Reply:
yes, we can use `is_floating_dtype`. I used a context knowledge that grid should have float dtype